autosklearn/util/progress_bar.py

```diff
@@ -10,59 +12,82 @@
 class ProgressBar(Thread):
     """A Thread that displays a tqdm progress bar in the console.
 
-    It is specialized to display information relevant to fitting to the training data
-    with auto-sklearn.
+    Treat this class as an ordinary thread. So to display a progress bar,
+    call start() on an instance of this class. To wait for the thread to
+    terminate call join(), which will max out the progress bar,
+    therefore terminate this thread immediately.
 
     Parameters
     ----------
     total : int
-        The total amount that should be reached by the progress bar once it finishes
-    update_interval : float
-        Specifies how frequently the progress bar is updated (in seconds)
-    disable : bool
-        Turns on or off the progress bar. If True, this thread won't be started or
-        initialized.
-    kwargs : Any
+        The total amount that should be reached by the progress bar once it finishes.
+    update_interval : float, default=1.0
+        Specifies how frequently the progress bar is updated (in seconds).
+    disable : bool, default=False
+        Turns on or off the progress bar. If True, this thread does not get
+        initialized and won't be started if start() is called.
+    tqdm_kwargs : Any, optional
         Keyword arguments that are passed into tqdm's constructor. Refer to:
-        `tqdm <https://tqdm.github.io/docs/tqdm/>`_. Note that postfix can not be
-        specified in the kwargs since it is already passed into tqdm by this class.
+        `tqdm <https://tqdm.github.io/docs/tqdm/>`_ for a list of parameters that
+        tqdm accepts. Note that 'postfix' cannot be specified in the kwargs since it is
+        already passed into tqdm by this class.
+
+    Examples
+    ----------
+        progress_bar = ProgressBar(
+            total=10,
+            desc="Executing code that runs for 10 seconds",
+            colour="green",
+        )
+        # colour is a tqdm parameter passed as a tqdm_kwargs
+
+        try:
+            progress_bar.start()
+
+            # some code that runs for 10 seconds
+
+        except SomeException:
+            # something went wrong
+        finally:
+            progress_bar.join()
+
+            # perform some cleanup
```

Comment:
My guess is that this won't render as expected. Two things, namely that sphinx-rst is overly sensitive to the length of the dashes `---`

```rst
Examples
------------
Whoops

Examples
--------
Yay
```

Secondly, it's also a bit anal about spacing for code blocks and won't have syntax highlighting by just using an indent.

This isn't in public api and doesn't get rendered so I don't think it's too necessary
All in all it should likely look something like

```rst
Examples
--------

.. code:: python

        progress_bar = ProgressBar(
            total=10,
            desc="Executing code that runs for 10 seconds",
            colour="green",
        )
        # colour is a tqdm parameter passed as a tqdm_kwargs
        try:
            progress_bar.start()
            # some code that runs for 10 seconds
        except SomeException:
            # something went wrong
        finally:
            progress_bar.join()
            # perform some cleanup
```

I'm fairly certain that the spacing between the header and the code block part are necessary.... sigh, you can see why we should move away from this

Reply:
Just pushed the updated doc.
Thanks you for the useful tips and for correcting it!